packages/webview_flutter/webview_flutter_android/lib/webview_android_widget.dart

```diff
@@ -0,0 +1,632 @@
+  }
+
+  @override
+  Future<String> runJavascriptReturningResult(String javascript) async {
+    return await webView.evaluateJavascript(javascript) ?? '';
+  }
+
+  @override
+  Future<void> addJavascriptChannels(Set<String> javascriptChannelNames) {
+    return Future.wait(
+      javascriptChannelNames.where(
+        (String channelName) {
+          return !_javaScriptChannels.containsKey(channelName);
+        },
+      ).map<Future<void>>(
+        (String channelName) {
+          final WebViewAndroidJavaScriptChannel javaScriptChannel =
+              WebViewAndroidJavaScriptChannel(
+                  channelName, javascriptChannelRegistry);
+          _javaScriptChannels[channelName] = javaScriptChannel;
+          return webView.addJavaScriptChannel(javaScriptChannel);
+        },
+      ),
+    );
+  }
+
+  @override
+  Future<void> removeJavascriptChannels(
+    Set<String> javascriptChannelNames,
+  ) {
+    return Future.wait(
+      javascriptChannelNames.where(
+        (String channelName) {
+          return _javaScriptChannels.containsKey(channelName);
+        },
+      ).map<Future<void>>(
+        (String channelName) {
+          final WebViewAndroidJavaScriptChannel javaScriptChannel =
+              _javaScriptChannels[channelName]!;
+          _javaScriptChannels.remove(channelName);
+          return webView.removeJavaScriptChannel(javaScriptChannel);
+        },
+      ),
+    );
+  }
+
+  @override
+  Future<String?> getTitle() => webView.getTitle();
+
+  @override
+  Future<void> scrollTo(int x, int y) => webView.scrollTo(x, y);
+
+  @override
+  Future<void> scrollBy(int x, int y) => webView.scrollBy(x, y);
+
+  @override
+  Future<int> getScrollX() => webView.getScrollX();
+
+  @override
+  Future<int> getScrollY() => webView.getScrollY();
+
+  Future<void> _release() => webView.release();
+
+  void _setCreationParams(CreationParams creationParams) {
+    final WebSettings? webSettings = creationParams.webSettings;
+    if (webSettings != null) {
+      updateSettings(webSettings);
+    }
+
+    final String? userAgent = creationParams.userAgent;
+    if (userAgent != null) {
+      webView.settings.setUserAgentString(userAgent);
+    }
+
+    final AutoMediaPlaybackPolicy autoMediaPlaybackPolicy =
+        creationParams.autoMediaPlaybackPolicy;
+    switch (autoMediaPlaybackPolicy) {
+      case AutoMediaPlaybackPolicy.always_allow:
+        webView.settings.setMediaPlaybackRequiresUserGesture(false);
+        break;
+      default:
+        webView.settings.setMediaPlaybackRequiresUserGesture(true);
+    }
+
+    addJavascriptChannels(creationParams.javascriptChannelNames);
+  }
+
+  Future<void> _trySetHasProgressTracking(bool? hasProgressTracking) {
+    if (hasProgressTracking == true) {
+      webChromeClient._onProgress = callbacksHandler.onProgress;
+    } else if (hasProgressTracking == false) {
+      webChromeClient._onProgress = null;
+    }
+
+    return Future<void>.sync(() => null);
```

Comment:
All the other functions use an early return for null; please be consistent to make understanding the batch of methods easier.